### PR TITLE
[Tables] updating recordings for latest breaking change

### PR DESCRIPTION
**Pull request by seankane-msft**
This is entirely recording files and go.mod/sum changes. No code changes, only recording.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -117,6 +117,11 @@ stages:
           EnvVars:
             AZURE_RECORD_MODE: 'playback'
 
+      - pwsh: |
+          docker logs ambitious_azsdk_test_proxy
```
Comment (benbp) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:121]: Is this getting run twice now?

Comment (seankane-msft) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Fixed

**Review by chlowell — APPROVED** [review]

File: eng/pipelines/templates/steps/build-test.yml [hunk 2]
```diff
@@ -60,8 +60,9 @@ steps:
 
   - ${{ if eq(parameters.TestProxy, true) }}:
     - pwsh: |
-        # $(Build.SourcesDirectory)/test-proxy.log is the hardcoded output log location for the test-proxy-tool.yml
-        cat $(Build.SourcesDirectory)/test-proxy.log
+        # ambitious_azsdk_test_proxy is the hardcoded container name used
+        # by the test proxy startup script
+        docker logs ambitious_azsdk_test_proxy
```
Comment (scbedd) [inline eng/pipelines/templates/steps/build-test.yml:65]: @seankane-msft https://github.com/Azure/azure-sdk-tools/issues/2539 is still open and I haven't had time to get to it. I disagree with re-enabling docker here, and you're seeing a bunch of builds fail already. 

You're already seeing a bunch of failures unfortunately.

Comment (seankane-msft) [reply eng/pipelines/templates/steps/build-test.yml:65]: #16921 is reverting the change

**Pull request merged**